Sources/App/Controllers/PackageController+routes.swift

```diff
@@ -380,9 +381,9 @@ extension PackageController {
         let baseURL = "http://\(baseURLHost)/\(baseURLPath)"
 
         switch fragment {
-            case .css, .data, .documentationRedirect, .documentation, .img, .index, .js:
+            case .css, .data, .documentation, .img, .index, .js:
```

Comment:
I'm not sure how this actually worked but the URL generated by `documentationRedirect` are definitely incorrect:

```
http://awsDocsBucket.s3-website.us-east-2.amazonaws.com/owner/package/1.2.3/documentationRedirect/
```

Instead, I'm using

```
let isDocumentationRedirect = (fragment == .documentation && archive == nil)
```

to detect the redirect. I _think_ that's what the want but TBH that whole function `documentation(req:fragment)` has grown way too big.

See https://github.com/SwiftPackageIndex/SwiftPackageIndex-Server/pull/1945/files#r945803903

Reply:
Agree completely that this funcrtion is out of control! 😬 I considered breaking it up several times during the versioned documentation changes, but there's no clear best approach to that.

The documentationRedirect is one of those "one-way" routes we were talking about this morning. Only ever use incoming, we never use it to generate a URL.

Reply:
I think at least everything inside the `case .documentation` should be a function, something we can also test in isolation. There might be further ways to break this down and it might make sense to do this before fiddling with `SiteURL`.

Reply:
I won't do this as part of this PR. That's logic that's happening after routing so it doesn't really need to be tested for the purpose of switching out the routers.